packages/expo-contacts/ios/ContactPickerControllerDelegate.swift

```diff
@@ -0,0 +1,27 @@
+import ContactsUI
+import ExpoModulesCore
+
+protocol OnContactPickingResultHandler {
+  func didPickContact(contact: CNContact) throws
+  func didCancelPickingContact()
+}
+
+class ContactPickerControllerDelegate: NSObject, CNContactPickerDelegate {
+  private let onContactPickingResultHandler: OnContactPickingResultHandler
+
+  init(onContactPickingResultHandler: OnContactPickingResultHandler) {
+    self.onContactPickingResultHandler = onContactPickingResultHandler
+  }
+
+  func contactPicker(_ picker: CNContactPickerViewController, didSelect contact: CNContact) {
+    do {
+      try self.onContactPickingResultHandler.didPickContact(contact: contact)
+    } catch {}
```

Comment:
Looking at this now, we'll need to deal with this error and clear the promise after. Otherwise it will fail, and attempting to try again will also fail because the promise won't be nil